docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,272 @@
+Domyślnie wartość flagi wynosi `false`, co oznacza, że konfiguracja pobierana jest z pliku `kubeconfig`.
+W przypadku, gdy flaga przyjmuje wartość `true`, konfiguracja pobierana jest bezpośrednio z klastra Kubernetes.
+
+Lokalizacja pliku `kubeconfig` zależy od dwóch wartości:
+- Flagi `--kubeconfig`- ścieżka do pliku `kubeconfig`.
+- Zmiennej środowiskowej `KUBECONFIG`- ścieżka do pliku `kubeconfig`.
+
+Jeżeli żadna z tych wartości nie jest podana, domyślnie aplikacja korzysta z pliku `~/.kube/config`.
+
+### Funkcje pomocnicze
+- **GetResourceGroupVersion**- na podstawie typu zasobu zwraca informacje potrzebne do generycznego wywołania funkcji `client-go`.
+  Dodatkowo zwraca informacje o tym, czy zasób jest namespace'owany, czy nie.
+- **GetResourceInterface**- dodatkowy poziom abstrakcji umożliwiający wstrzykiwanie zależności (dependency injection) w testach jednostkowych.
+  Równocześnie upraszcza i ujednolica wywołanie funkcji `client-go`.
+- **WatchForChanges**- funkcja umożliwiająca obserwowanie zmian w zasobie opisującym role wykorzystywane w autoryzacji.
+- **transposeResourceListColumns**- funkcja pomocnicza, która zamienia listę zasobów i ich pól (postaci mapy nazwa_zasobu -> jej_pola) na listę pól i do jakich typów zasobów należą (postaci mapy nazwa_pola -> lista_typów).
+
+### Funkcje CreateResource, GetResource, UpdateResource, DeleteResource
+Funkcje te wykorzystują bibliotekę `client-go` do komunikacji z API klastra Kubernetes. Jako argumenty przyjmują kolejno:
+- GetResource- typ zasobu (resourceType), namespace, nazwę zasobu (resourceName).
+- CreateResource- typ zasobu (resourceType), namespace, zasób (ResourceDetails).
+- UpdateResource- typ zasobu (resourceType), namespace, nazwę zasobu (resourceName), zasób (ResourceDetails).
+- DeleteResource- typ zasobu (resourceType), namespace, nazwę zasobu (resourceName).
+
+W przypadku gdy zasób jest namespace'owany, a użytkownik nie poda namespace, zasób zostanie utworzony w namespace'u `default`.
+
+ResourceDetails to luźna struktura, która reprezentuje nieustrukturyzowane dane zasobu.
+
+Funkcje create, get i update zwracają zasób w postaci struktury `ResourceDetails` lub błąd w przypadku niepowodzenia. Funkcja delete zwraca błąd w przypadku niepowodzenia.
+
+Funkcje te są w stanie obsłużyć wszystkie typy zasobów, jednak ze względu na funkcje listowania zakres został ograniczony do 20 typów wymienionych w części `Funkcja list`.
+
+### Funkcja ListResources
+Funkcja ta wykorzystuje bibliotekę `client-go` do komunikacji z API klastra Kubernetes. 
+Jako argumenty przyjmuje: typ zasobu (resourceType), namespace.
+
+W przypadku gdy zasób jest namespace'owany, a użytkownik nie poda namespace, zwracane są zasoby ze wszystkich namespace'ów.
+
+Funkcja dla każdego typu zasobu zwraca określoną listę wartości oraz listę nazw zwracanych pól.
+- Dla zasobów typu `ReplicaSet` zwracane są wartości `name`, `namespace`, `desired`, `current`, `ready`, `age`.
```

Comment:
Nie wiem czy bym to wypisywał 1 do 1 wszystko bo nikt nie będzie pamiętał o aktualizacji tego wszystkiego. Jakoś bym generycznie napisał, że są konkretne ustalone wartości dla każdego z zasobów z osobna i tyle

Reply:
Myślę, że to powinno być gdzieś w dokumentacji, a lepszego miejsca nie widzę.